akka-actor-typed-tests/src/test/scala/akka/actor/typed/WatchSpec.scala

```diff
@@ -66,5 +74,45 @@ class WatchSpec extends TestKit("WordSpec")
 
       receivedTerminationSignal.future.futureValue shouldEqual CustomTerminationMessage
     }
+
+    class ErrorTestSetup {
+      val terminator = systemActor(terminatorBehavior)
+
+      val watcher = systemActor(Behaviors.immutable[Message] {
+        case (ctx, StartWatchingWith(watchee, msg)) ⇒
+          ctx.watchWith(watchee, msg)
+          Behaviors.same
+        case (ctx, StartWatching(watchee)) ⇒
+          ctx.watch(watchee)
+          Behaviors.same
+        case (_, msg) ⇒
+          Behaviors.stopped
+      })
+    }
+    "warn when watch is used after watchWith on same subject" in new ErrorTestSetup {
+      watcher ! StartWatchingWith(terminator, CustomTerminationMessage)
+
+      EventFilter[IllegalStateException](pattern = ".*termination message was not overwritten.*", occurrences = 1) intercept {
+        watcher ! StartWatching(terminator)
+      }
```

Comment:
also verify that it was stopped? it throws exception right?

Reply:
Done

Reply:
Just FYI there is a `TestProbe.expectTerminated` now.